.github/workflows/planner_tests.yaml

```diff
@@ -0,0 +1,102 @@
+name: Planner Integration Tests
+on:
+  pull_request:
+  workflow_call:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read
+
+jobs:
+  detect-changes:
+    runs-on: ubuntu-latest
+    outputs:
+      planner: ${{ steps.filter.outputs.planner }}
+    steps:
+      - uses: actions/checkout@v5
+      - uses: dorny/paths-filter@v3
+        id: filter
+        with:
+          filters: |
+            planner:
+              - 'planner/**'
+              - 'internal/planner/**'
+              - '.github/workflows/planner_tests.yaml'
+  integration-test:
+    needs: detect-changes
+    if: needs.detect-changes.outputs.planner == 'true'
+    runs-on: [self-hosted-linux-amd64-noble-edge]
+    name: Run Integration Tests
+    steps:
+      - uses: actions/checkout@v5
+
+      - name: Set up Go
+        uses: actions/setup-go@v6
+        with:
+          go-version: 1.25
+
+      - name: Build and Test
+        env:
+          POSTGRESQL_DB_CONNECT_STRING: postgres://postgres:postgres@localhost:5432/postgres
+          POSTGRESQL_DB_PORT: 8080
```

Comment:
Is this DB port correct? Since other part is referring to the 5432 port.

Reply:
i think this has the same purpose as APP_PORT defined in webhook tests. I was trying to follow https://documentation.ubuntu.com/charmcraft/latest/howto/manage-web-app-charms/integrate-web-app-charm/#integrate-with-a-database hence the naming